cmd/cdi-controller/controller.go

```diff
@@ -101,10 +111,16 @@ func start(cfg *rest.Config, stopCh <-chan struct{}) {
 	serviceInformerFactory := k8sinformers.NewFilteredSharedInformerFactory(client, common.DefaultResyncPeriod, "", func(options *v1.ListOptions) {
 		options.LabelSelector = common.CDILabelSelector
 	})
+	ingressInformerFactory := k8sinformers.NewFilteredSharedInformerFactory(client, common.DefaultResyncPeriod, "", func(options *v1.ListOptions) {
+		options.LabelSelector = common.CDILabelSelector
+	})
+	routeInformerFactory := routeinformers.NewSharedInformerFactory(openshiftClient, common.DefaultResyncPeriod)
```

Comment:
Don't we want to call isOpenShift before setting this,as you call it before you start routeInformerfactory later on...

Reply:
it doesn't work until we run the go routine

Reply:
I know. My intention was why we set it unconditionally, where we might not execute the go routine...